### ISSUE TITLE
HLS reports non-existent errors when using Calamity and Polysemy with Polysemy.Plugin (polysemy-plugin)

### ISSUE THREAD

**Issue by MonAaraj** [haskell/haskell-language-server#2989]
haskell-language-server-wrapper --probe-tools:
```
haskell-language-server version: 1.7.0.0 (GHC: 9.2.2) (PATH: /home/mon/.local/share/ghcup/hls/1.7.0.0/lib/haskell-language-server-1.7.0.0/bin/haskell-language-server-wrapper)
Tool versions found on the $PATH
cabal:          3.6.2.0
stack:          Not found
ghc:            9.2.2
```
(I launched HLS in the terminal with `haskell-language-server-wrapper` to provide the errors it reports in the paste below)
HLS reports errors that seem to not exist, because building & running the project works, and testing the program results in correct behaviour. I have this paste: https://bpa.st/J32A which has the Main.hs and the errors that HLS thinks exist. The git repository is provided later on in this issue.

### Your environment

Which OS do you use:
Arch Linux (Artix variant)
Which LSP client (editor/plugin) do you use:
doom emacs+lsp-mode
Describe your project (alternative: link to the project): https://git.sr.ht/~monaaraj/bot

### Steps to reproduce
I just run doom emacs and open up my project. This issue happens only with this calamity and polysemy (with polysemy-plugin) project, HLS acts well in every other project I have.

### Expected behaviour
The program should be deemed without error, if GHC builds and runs the program, with the program successfully running without issue.

### Actual behaviour
HLS reports very very weird errors regarding Internal.Typeable things, P.Member instance overlaps, ambiguous use of type variables... the specific errors have been provided in the bpaste link above.

### Include debug information

The HLS log seems to be too long to be pasted to any paste service, so I have provided it as an attachment file:
[hls.log](https://github.com/haskell/haskell-language-server/files/8985976/hls.log)

**Comment by MorrowM** [comment]
It's worth pointing out that these are the exact error messages you get from `polysemy-plugin` not being enabled.

**Comment by MorrowM** [comment]
It appears this is 9.2 specific. On 8.10.7 and 9.0.2 HLS does not raise any of the errors.

**Comment by oberblastmeister** [comment]
This also happens with [effectful-plugin](https://github.com/haskell-effectful/effectful/tree/master/effectful-plugin)

**Comment by michaelpj** [comment]
I think @pepeiborra did the 9.2 compatibility work? Did something change wrt plugins?

**Comment by sonowz** [comment]
I'm having same issue with `polysemy-plugin`, and I strongly suggest that HLS does not handle GHC option `-fplugin=Polysemy.Plugin` correctly. GHC shows the same error as HLS if `-fplugin=Polysemy.Plugin` option is not given.

Can anyone confirm this behavior?

**Comment by tek** [comment]
@sonowz that is correct. interestingly, I've been observing this kind of misbehaviour ever since, but only very infrequently. With GHC 9.2, it is now permanent.

**Comment by PPKFS** [comment]
Completely forgot about this, but I had this issue with `Cleff.Plugin` some months back. 9.0.x was fine, but any 9.2.x build it would throw up HLS errors.

Ran into it again today when using `Effectful.Plugin` on 9.2. Again, vanishes if I swap to 9.0. It seems hls on 9.2 simply does not read plugins somehow.

**Comment by tek** [comment]
same, everything's working now

**Comment by michaelpj** [comment]
Yep, this should have been fixed in 1.9.0.0.
